flagx.go

```diff
@@ -350,10 +380,96 @@ func (f *FlagSet) Float64Set(p *map[float64]struct{}, name, alias string, value
 // The flag accepts a value acceptable to time.ParseDuration.
 // Empty string for alias means no alias will be created.
 func (f *FlagSet) DurationSet(p *map[time.Duration]struct{}, name, alias string, value map[time.Duration]struct{}, usage string) {
+	f.aliases[name] = alias
 	var sd SetOfDuration = value
 	*p = map[time.Duration]struct{}(sd)
 	f.fs.Var(&sd, name, usage)
 	if alias != "" {
 		f.fs.Var(&sd, alias, usage)
 	}
 }
+
+// PrintDefaults prints, to standard error unless configured otherwise, the
+// default values of all defined command-line flags in the set.
+func (f *FlagSet) PrintDefaults() {
+	// NOTE(junk1tm): copy-pasted from flag.PrintDefaults with a few modifications to support aliases.
+
+	var isZeroValueErrs []error
+	f.VisitAll(func(fl *flag.Flag) {
+		if _, ok := f.aliases[fl.Name]; !ok {
+			// The flag is an alias, do not print it separately.
+			return
+		}
+		var b strings.Builder
+		fmt.Fprintf(&b, "  -%s", fl.Name) // Two spaces before -; see next two comments.
+		if alias := f.aliases[fl.Name]; alias != "" {
+			fmt.Fprintf(&b, " (-%s)", alias)
+		}
+		name, usage := flag.UnquoteUsage(fl)
+		if len(name) > 0 {
+			b.WriteString(" ")
+			b.WriteString(name)
+		}
+		// Boolean flags of one ASCII letter are so common we
+		// treat them specially, putting their usage on the same line.
+		if b.Len() <= 4 { // space, space, '-', 'x'.
+			b.WriteString("\t")
+		} else {
+			// Four spaces before the tab triggers good alignment
+			// for both 4- and 8-space tab stops.
+			b.WriteString("\n    \t")
+		}
+		b.WriteString(strings.ReplaceAll(usage, "\n", "\n    \t"))
+
+		// Print the default value only if it differs to the zero value
+		// for this flag type.
+		if isZero, err := isZeroValue(fl, fl.DefValue); err != nil {
+			isZeroValueErrs = append(isZeroValueErrs, err)
+		} else if !isZero {
+			// HACK(junk1tm): flag.stringValue is unexported, so we have to compare the type's name.
+			if fmt.Sprintf("%T", fl.Value) == "*flag.stringValue" {
```

Comment:
@cristaloleg I'm not sure there is a better way :( We could also use reflect.KindOf to make sure it's a string, but technically it would be a different check

Reply:
Nah, leave it ;)